app/src/main/java/com/worldsnas/starplayer/model/MusicRepositoryImpl.kt

```diff
@@ -16,21 +19,48 @@ class MusicRepositoryImpl @Inject constructor(
             val apiMusicList = webServiceApi.getMusics(page, count)
 
             apiMusicList.body()?.map {
-
                 MusicRepoModel(
                     it.id.toInt(),
                     it.name,
                     it.artist,
                     "album",
                     "genre",
-                    it.musicLink
+                    it.musicLink,
+                    false
                 )
             }!!
         }
 
-
     override suspend fun getLocalData(): List<MusicRepoModel> =
         withContext(Dispatchers.IO) {
-            localMusicProvider.getAllMusic()
+            val localList = localMusicProvider.getAllMusic().toMutableList()
+            val favoritesList = favoriteMusicDao.getAllMusics()
+
+            for (localItem in localList) {
+                for (favItem in favoritesList) {
+                    if (localItem.id == favItem.id)
+                        localItem.isFavorite = true
+                }
+            }
```

Comment:
afarin in khob shode
vali benazaram behtare baad as in khat `DB` ro update konim ke ezafat hazf she masalan age music i favorite shode vali vojod nadare niazi nist to `DB` ham bashe

Reply:
ino nafahmidam, yani mazooreton music haye online hast?

Reply:
na shayad karbar music i ro az ro goshish hazf kone khob
ma nabayd DB mono alaki por konim (albate doroste be karbar neshon nemidim

Reply:
mishe har bar keh shart if bargharar bood item az favoritesList hazf kard, ba item haye baghi az DB delete kard.